src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -252,6 +477,215 @@ private TemporaryPoint createNewPoint(final Coordinate coordinate,
+        {
+            final LinearRing inner = innerRings.get(innerIndex);
+            boolean foundEnclosingOuter = false;
+            for (int outerIndex = 0; outerIndex < outerRings.size(); outerIndex++)
+            {
+                final LinearRing outer = outerRings.get(outerIndex);
+                final com.vividsolutions.jts.geom.Polygon outerPolygon = new com.vividsolutions.jts.geom.Polygon(
+                        outer, null, JtsUtility.GEOMETRY_FACTORY);
+                if (outerPolygon.contains(inner))
+                {
+                    foundEnclosingOuter = true;
+                    outerToInnerMap.add(outerIndex, innerIndex);
+                }
+            }
+
+            if (!foundEnclosingOuter)
+            {
+                // Isolated inner ring, invalid multipolygon
+                logger.error("Found isolated inner member for Multipolygon Relation geometry: {}",
+                        inner);
+            }
+        }
+    }
+
+    /**
+     * @param closedOuterLines
+     *            The list of closed outer {@link Line}s to use
+     * @param closedInnerLines
+     *            The list of closed inner {@link Line}s to use
+     * @return a mapping of intersecting outer to inner relation members
+     */
+    private MultiMap<Integer, Integer> findIntersectingMembers(final List<Line> closedOuterLines,
+            final List<Line> closedInnerLines)
+    {
+        final MultiMap<Integer, Integer> outerToInnerIntersectionMap = new MultiMap<>();
+
+        if (!closedOuterLines.isEmpty() && !closedInnerLines.isEmpty())
+        {
+            for (int innerIndex = 0; innerIndex < closedInnerLines.size(); innerIndex++)
+            {
+                final Line inner = closedInnerLines.get(innerIndex);
+                for (int outerIndex = 0; outerIndex < closedOuterLines.size(); outerIndex++)
+                {
+                    final Line outer = closedOuterLines.get(outerIndex);
+
+                    // Make sure we're looking at the same country, not just intersection
+                    if (fromSameCountry(outer, inner) && outer.intersects(new Polygon(inner)))
+                    {
+                        outerToInnerIntersectionMap.add(outerIndex, innerIndex);
+                    }
+                }
+            }
+        }
+
+        return outerToInnerIntersectionMap;
+    }
+
+    /**
+     * @param one
+     *            The first {@link Line}
+     * @param two
+     *            The second {@link Line}
+     * @return {@code true} if both lines belong to the same country
+     */
+    private boolean fromSameCountry(final Line one, final Line two)
+    {
+        final Optional<IsoCountry> countryOne = ISOCountryTag.first(one);
+        final Optional<IsoCountry> countryTwo = ISOCountryTag.first(two);
+        if (countryOne.isPresent() && countryTwo.isPresent())
+        {
+            return countryOne.get().equals(countryTwo.get());
+        }
+        return false;
```

Comment:
So what happens if we neither `one` or `two` don't have ISO country codes?

Reply:
Impossible, at this point, there's only 3 Atlas entities - Points, Lines and Relations. Relations are the only ones that haven't been assigned a country code yet, everything else has. I'll add an error condition just in case!